clients/TypeScript/packages/client/src/Connection.ts

```diff
@@ -0,0 +1,59 @@
+import WebSocket from 'isomorphic-ws'
+
+export interface ConnectionConfig {
+  host?: string,
+  port?: number
+}
+
+export interface InteractionContext {
+  socket?: WebSocket
+  connection?: ConnectionConfig
+  closeOnCompletion?: boolean
+  connectionString?: string
+}
+
+export type Mirror = { [k: string]: unknown }
+
+const createConnectionString = (connection?: ConnectionConfig): InteractionContext['connectionString'] =>
+  `ws://${connection?.host || 'localhost'}:${connection?.port || 1337}`
```

Comment:
It should eventually be allowed to connect to a WebSocket through TLS; thus making the protocol part of the `ConnectionConfig` with a sensible default perhaps :thinking: ?

Reply:
https://github.com/KtorZ/cardano-ogmios/pull/29/commits/e234fd2ff8009f714a10a52f51af0abaf8d06acd